components/SiteNavigationDisclosure.vue

```diff
@@ -0,0 +1,101 @@
+<template>
+  <Disclosure
+    as="div"
+  >
+    <DisclosureButton
+      class="group mt-2 w-full flex items-center pl-2 pr-1 py-2 space-x-2 text-left rounded-md hover:bg-gray-200 focus:outline-none focus:ring-2 focus:ring-inset ring-focus-ring"
+      :class="currentPath.startsWith(navigationItem.path) ? 'bg-gray-100' : ''"
+      @click="open = !open"
+    >
+      <component
+        :is="navigationItem.icon"
+        class="icon-md flex-shrink-0 text-content-tertiary group-hover:text-content-secondary"
+        aria-hidden="true"
+      />
+      <span class="flex-1 text-content-primary text-base leading-5 font-medium">{{ navigationItem.title }}</span>
+      <disclosure-arrow-right :class="[open ? 'text-gray-400 rotate-90' : 'text-gray-300', 'ml-3 icon-md flex-shrink-0 transform transition-colors duration-150 ease-in-out group-hover:text-gray-400']" />
+    </DisclosureButton>
+    <transition
+      enter-active-class="transition duration-100 ease-out"
+      enter-from-class="transform scale-95 opacity-0"
+      enter-to-class="transform scale-100 opacity-100"
+      leave-active-class="transition duration-75 ease-out"
+      leave-from-class="transform scale-100 opacity-100"
+      leave-to-class="transform scale-95 opacity-0"
+    >
+      <div v-show="open">
+        <DisclosurePanel static>
+          <div
+            v-for="firstChild in navigationItem.children"
+            :key="firstChild.title"
+          >
+            <NuxtLink
+              :to="firstChild.path"
+              class="group mt-2 flex w-full items-center rounded-md py-2 pl-6 pr-2 text-sm font-medium text-content-secondary hover:bg-gray-200 hover:text-gray-900 focus:outline-none focus:ring-2 focus:ring-inset ring-focus-ring"
+              @click="$emit('link-clicked')"
+            >
+              {{ firstChild.title }}
+            </NuxtLink>
+            <div v-if="firstChild.children">
+              <div
+                v-for="secondChild in firstChild.children"
+                :key="secondChild.title"
+              >
+                <NuxtLink
+                  :to="secondChild.path"
+                  class="group mt-2 flex w-full items-center rounded-md py-2 pl-8 pr-2 text-sm font-medium text-gray-600 hover:bg-gray-200 hover:text-gray-900 focus:outline-none focus:ring-2 focus:ring-inset ring-focus-ring"
+                  @click="$emit('link-clicked')"
+                >
+                  {{ secondChild.title }}
+                </NuxtLink>
+              </div>
+            </div>
+          </div>
+        </DisclosurePanel>
+      </div>
+    </transition>
+  </Disclosure>
+</template>
+
+<script setup>
+import { ref } from 'vue';
+import { useRoute } from 'vue-router';
+
+import {
+  Disclosure,
+  DisclosureButton,
+  DisclosurePanel,
+} from '@headlessui/vue';
+
+const props = defineProps({
+  navigationItem: { type: Object, required: true },
+  urlToActiveNav: { type: Object, required: true },
```

Comment:
I decided to pass this through as a prop so the manually configured nav is all in one place.